### PR TITLE
[CHK-1139] add policy to track data for log purpose

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Edit and add policy for ecommerce api, to put rptIds and transactionId values in the header for log purpose

### List of changes

Added policy file for read multiple rptiIds in transaction activate.
Edit jwt validation token file to put transactionId in the header

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/core/apim_checkout.tf [hunk 1]
```diff
@@ -346,3 +346,12 @@ resource "azurerm_api_management_api_operation_policy" "transaction_authorizatio
     ecommerce_vpos_psps_list = var.ecommerce_vpos_psps_list
   })
 }
+
+resource "azurerm_api_management_api_operation_policy" "transaction_activation_request" {
```
Comment (infantesimone) [inline src/core/apim_checkout.tf:350]: to delete, because they refer to the old deprecated APIs

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf [hunk 2]
```diff
@@ -100,7 +100,7 @@ resource "azurerm_api_management_api_operation_policy" "delete_transaction" {
   xml_content = file("./api/ecommerce-checkout/v1/_validate_transactions_jwt_token.tpl")
 }
 
-resource "azurerm_api_management_api_operation_policy" "get_payment_request_info_api" {
+resource "azurerm_api_management_api_operation_policy" "get_payment_request_info_api_policy" {
```
Comment (infantesimone) [inline src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf:103]: to update xml content with _payment_request_policy.xml.tpl

File: src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf [hunk 3]
```diff
@@ -132,3 +132,12 @@ resource "azurerm_api_management_api_operation_policy" "get_carts_redirect" {
   })
 }
 
+resource "azurerm_api_management_api_operation_policy" "transaction_activation_request" {
+  api_name            = "${local.project}-checkout-ecommerce-api-v1"
```
Comment (infantesimone) [inline src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf:136]:
```suggestion
  api_name            = "${local.project}-ecommerce-checkout-api-v1"
```

File: src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf [hunk 4]
```diff
@@ -132,3 +132,12 @@ resource "azurerm_api_management_api_operation_policy" "get_carts_redirect" {
...
+  api_management_name = module.apim.name
+  resource_group_name = azurerm_resource_group.rg_api.name
```
Comment (infantesimone) [inline src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf:138]:
```suggestion
  resource_group_name = local.pagopa_apim_rg

File: src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf (hunk 2 above)
Comment (ciuffagianluca) [reply src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf:103]: where's my head? :D

File: src/core/apim_checkout.tf (hunk 1 above)
Comment (ciuffagianluca) [reply src/core/apim_checkout.tf:350]: All this file is to delete or only this resource?

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf [hunk 5]
```diff
@@ -132,3 +132,12 @@ resource "azurerm_api_management_api_operation_policy" "get_carts_redirect" {
   })
 }
 
+resource "azurerm_api_management_api_operation_policy" "transaction_activation_request" {
```
Comment (infantesimone) [inline src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf:135]: TODO rollback all the changes in the core folder

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf [hunk 6]
```diff
@@ -132,3 +132,12 @@ resource "azurerm_api_management_api_operation_policy" "get_carts_redirect" {
...
+  api_name            = "${local.project}-ecommerce-checkout-api-v1"
+  api_management_name = module.apim.name
+  resource_group_name = local.pagopa_apim_rg
+  operation_id        = "newTransaction"
+
+  xml_content = file("./api/checkout/checkout_ecommerce/v1/_rpt_ids_value.xml.tpl")
```
Comment (infantesimone) [inline src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf:141]: this path is wrong, maybe  `api/ecommerce-checkout/v1` ?

Comment (ciuffagianluca) [reply src/domains/ecommerce-app/04_apim_ecommerce_checkout.tf]: right path set

**Pull request merged**